### PR TITLE
Lhvm integration

### PR DESCRIPTION
**Pull request by Daniels118**
- Added *ScriptHeaders* folder with enums used by CHL scripts;
- Added the `CHLApi` class, which provides a functions table used to initialize the VM. It comes with the skeleton for all the 464 functions (POP of inputs and PUSH of outputs with the proper types). The `CHLApi` can be accessed through the `Locator` service.
- implemented some functions to allow the execution of basic CHL scripts (running other scripts, creating objects and moving them around).

Address issue https://github.com/openblack/openblack/issues/7

### CODE REVIEW

**Comment by bwrsandman** [comment]
You libcap problems are likely solved with https://github.com/openblack/openblack/pull/792

File: src/ScriptHeaders/ScriptEnumstwo.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/******************************************************************************
```
Comment (bwrsandman) [inline src/ScriptHeaders/ScriptEnumstwo.h:1]: Why this filename and not just add these to Enum.h?

File: src/CHLApi.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+class CHLApi
+{
+public:
+	CHLApi();
+
+	[[nodiscard]] const std::vector<lhvm::NativeFunction>& GetFunctionsTable() { return _functionsTable; };
+
+private:
+#define CREATE_FUNCTION_BINDING(NAME, STACKIN, STACKOUT, FUNCTION)       \
```
Comment (bwrsandman) [inline src/CHLApi.h:30]: You should move this to the .cpp

**Review by bwrsandman — CHANGES_REQUESTED** [review]
Small changes but I like what I see. 

One thing though, I'd add logging for the to-do

File: src/ScriptHeaders/ScriptEnumstwo.h (hunk 1 above)
Comment (Daniels118) [reply src/ScriptHeaders/ScriptEnumstwo.h:1]: There is no technical reason to have 2 distinct files, but the original scripting tools from LH have these 2, so keeping the same filenames make it easier to remember from where they came from. If you wish to maintain less files, I could merge them.

**Review by bwrsandman — APPROVED** [review]

**Pull request merged**